### PR TITLE
feat: implement ng add for store and effects packages

### PR DESCRIPTION
**Pull request by vitaliy-bobrov**
Implementation of #920 WIP.

This is the implementation of @ngrx/store ng-add module to get feedback, cause other `ng-add`s will be done in a similar way.

Done:
- created schematics "submodule" inside @ngrx/store with needed build configuration
- implemented optional addition `@ngrx/store` as `dependency` to `package.json` (idea from https://github.com/angular/material2/tree/master/src/lib/schematics)
- store `ng-add` doesn't import and initialize `StoreDevtoolsModule` (imho should be done ny `store-devtools` ng-add)

Additional code share optimizations could be performed later, after initial implementation.

Will work on other modules after gathering a feedback.

### CODE REVIEW

File: modules/store/schematics/tsconfig-build.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (brandonroberts) [inline modules/store/schematics/tsconfig-build.json]: Remove this file. The `npm_package` already has a default tsconfig.json` to use

File: modules/store/schematics/ng-add/index.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  addImportToModule,
+  buildRelativePath,
+  findModuleFromOptions,
+  getProjectPath,
+  insertImport,
+  stringUtils,
+  addPackageToPackageJson,
+  platformVersion,
+  parseName,
+} from '../../schematics-core';
```
Comment (brandonroberts) [inline modules/store/schematics/ng-add/index.ts]: Use the `@ngrx/schematics/schematics-core` import here

File: modules/store/schematics/ng-add/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  const projectPath = getTestProjectPath();
+
+  let appTree: UnitTestTree;
+
+  beforeEach(() => {
+    appTree = createWorkspace(schematicRunner, appTree);
+  });
+
+  it('should update package.json', () => {
```
Comment (brandonroberts) [inline modules/store/schematics/ng-add/index.spec.ts]: Add a test for the `skipPackageJson` option

File: modules/store/package.json [hunk 4]
```diff
@@ -19,8 +22,11 @@
   "homepage": "https://github.com/ngrx/platform#readme",
   "peerDependencies": {
     "@angular/core": "NG_VERSION",
+    "@angular-devkit/core": "NG_DEVKIT_VERSION",
```
Comment (brandonroberts) [inline modules/store/package.json]: Remove these from the `package.json` as we are treating them as implicit dependencies instead of adding additional dependencies to `@ngrx/store`

File: .prettierignore [hunk 5]
```diff
@@ -1,5 +1,7 @@
 /dist
 /modules/schematics/src/*/files/*
+/modules/schematics-core/templates/*
```
Comment (brandonroberts) [inline .prettierignore]: Change `templates` to `files` so its consistent with schematics. If we want to change `files` to `templates` across the board, we'll do that in a separate PR.

File: modules/store/schematics/BUILD [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (brandonroberts) [inline modules/store/schematics/BUILD]: This should be added to the `packages` section in the `modules/store/BUILD` file in the `npm_package`. That way its output gets copied into the distribution.

**Comment by brandonroberts** [comment]
I left some feedback. I'd rather have 2 copies of the store template files, then put them into the `schematics-core` and have 6 copies in places where they will never be used.

**Comment by vitaliy-bobrov** [comment]
Thanks for the review, will work on changes. Regarding templates, I also not sure about keeping them in core, just thought that it could be good to have the only source.

**Comment by brandonroberts** [comment]
Since its only one file, I think it will be ok to have 2 copies.

File: modules/store/package.json [hunk 7]
```diff
@@ -21,6 +24,7 @@
     "@angular/core": "NG_VERSION",
     "rxjs": "RXJS_VERSION"
   },
+  "schematics": "./schematics/collection.json",
```
Comment (brandonroberts) [inline modules/store/package.json]: Add this replacement to the `/tools/defaults.bzl` file

File: modules/store/schematics/ng-add/index.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      sourceText,
+      ts.ScriptTarget.Latest,
+      true
+    );
+
+    const statePath = `${options.path}/${options.statePath}`;
+    const relativePath = buildRelativePath(modulePath, statePath);
+    const environmentsPath = buildRelativePath(
+      statePath,
+      `/${options.path}/environments/environment`
```
Comment (brandonroberts) [inline modules/store/schematics/ng-add/index.ts]: Update this to match https://github.com/ngrx/platform/blob/master/modules/schematics/src/store/index.ts#L57

File: modules/store/schematics/ng-add/index.ts [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  addImportToModule,
+  buildRelativePath,
+  findModuleFromOptions,
+  getProjectPath,
+  insertImport,
+  stringUtils,
+  addPackageToPackageJson,
+  platformVersion,
+  parseName,
+} from '@ngrx/schematics/schematics-core';
```
Comment (brandonroberts) [inline modules/store/schematics/ng-add/index.ts]: This should be `@ngrx/store/schematics-core`

File: modules/store/package.json (hunk 7 above)
Comment (brandonroberts) [inline modules/store/package.json]: Add this replacement to `/tools/default.bzl`

**Comment by brandonroberts** [comment]
Added a couple more changes, otherwise I think this is ready to implement in the other packages.
